### ISSUE TITLE
partial F-droid inclusion

### ISSUE THREAD

**Issue by AnonymousSeal** [ProtonMail/proton-mail-android#61]
Many of the people, who use/advocate for/support Proton services the most are people who care about their privacy and a lot of them are Fdroid users. So a F-droid inclusion should really be a priority.

I know this isn't easy, but I have an idea of what you could do. It would be nice if you handled it like SimpleLogin (if I remember it correctly): They use Google Analytics in their Play Store App, but also have a foss solution for F-droid, where the use open source software to caption bug reports.

Maybe you could do the same thing with the notification system your using. Using Google Push Notifications in the Play Store version and a foss solution for a F-droid release, because the F-droid community probably cares more about the product not using google, than a slight increase in power usage. And the standard users will just pick the regular option, which is probably best suited for them.

Also I should mention that this issue is closely related to issue #7, but it provides a different idea so I decided to make this an independent issue.

**Comment by jpastuszek** [comment]
To be honest I personally don't care about push notifications for e-mail - I just check the inbox once a day manually.

I use microG on my phone. So I don't have Google Apps/Play but I can sill use Google's push notification though microG implementation. Many apps register with it fine and work correctly so that could be an option.

I would like to have ProtonMail app on f-droid even if it would not get notifications at all. Also the non-play store downloaded app does not support push notifications as well, so I don't see how this is a blocking issue for f-droid release.

In a perfect world we would have some FOSS service which you can self host and configure any instance address on you phone for push notifications for apps to use.

As you note, not having ProtonMail app on f-droid is a major factor when comparing privacy oriented e-mail providers for users aiming at deGoogling their life.

**Comment by laanwj** [comment]
I too would be happy with a no-push-notification ProtonMail app in F-droid.

**Comment by hefty-fading-culprit-epiphany-drew** [comment]
> I too would be happy with a no-push-notification ProtonMail app in F-droid.

+1

**Comment by ksdrummer** [comment]
I totally agree with this ! I'm concerned about privacy and that's the reason I chose protonmail. However, the fact that the app insn't in the f-droid store and that it uses google push notification made me lost the trust I had for protonmail.
How can protonmail advertise so much about privacy and just use google API like every other non-free app ? I think that's a shame

**Comment by laanwj** [comment]
If you're concerned about getting applications from the play store you could install the apk from the [releases section](https://github.com/ProtonMail/proton-mail-android/releases/tag/1.13.25). The only drawback (besides not having notifications) is that you won't have automatic updates.

**Comment by 4face-studi0** [comment]
> I totally agree with this ! I'm concerned about privacy and that's the reason I chose protonmail. However, the fact that the app insn't in the f-droid store and that it uses google push notification made me lost the trust I had for protonmail.
> How can protonmail advertise so much about privacy and just use google API like every other non-free app ? I think that's a shame

Being Android itself a Google product, FCM is the only push notification service integrated in the system and there are no other equally reliable solutions as now.

Having ProtonMail a large user-base and targeting also average users, it must provide the best user experience.
Remind that, FCM is integrated in the system, FCM library is only an interface for comunicate with it, if your device doesn't have Play Services, it will be a dead and harmless piece of code.

**Comment by ksdrummer** [comment]
@4face-studi0 since you ask, I use lineageOS with microG and so I don't have push notifications.

It is always the same debate : "better user experience vs privacy". Since protonmail is actively advertising about privacy I would expect they choose privacy.

If you think this way it is easy to conclude that if you want better UI just use google products. IMO, lots of foss and privacy based programs have pretty good UI so please stop using this argument.  

For now, I'm just feeling I got screwed by protomail adverting. My only hope is PM is saying they are working in a FCM alternative. Let's see

**Comment by 4face-studi0** [comment]
@ksdrummer ProtonMail aims to be a competitive Mail service with the best privacy.

Every user has different requirements, but a notification system is a core feature for 95% of the users, thus cannot be sacrificed.

**Comment by ksdrummer** [comment]
@4face-studi0 that's why they should use another one, just like others privacy based mail providers (I see tutanota is on f-droid). If they do it, PM should be able to to it too.

Using FCM doesn't makes PM a mail service using a good privacy at all.

**Comment by ksdrummer** [comment]
What's a competitive mail ? A mail which sends e-mails better than the other ?

**Comment by hefty-fading-culprit-epiphany-drew** [comment]
Let's be honest here, people caring about privacy know that privacy and Features can't exist at the the same time, so a partial f Droid inclusion would be very welcome from the community

**Comment by digital-mystik** [comment]
A good comprise would be to have a Proton F-Droid repo so that releases from upstream can be tracked/pushed/updated easily without using the Play Store. A number of projects have their own for this purpose such as Bitwarden etc.

**Comment by hefty-fading-culprit-epiphany-drew** [comment]
Once can literally fork this and remove the notification part and ask f Droid to build it, only if I knew coding 😂

**Comment by ksdrummer** [comment]
If only I knew coding I would host myself my email server and don't have to rely on a profit company which is ok to use google services :trollface:

**Comment by hefty-fading-culprit-epiphany-drew** [comment]
> If only I knew coding I would host myself my email server and don't have to rely on a profit company who is ok to use google services :trollface:

This says a lot about society

**Comment by AnonymousSeal** [comment]
> ksdrummer ProtonMail aims to be a competitive Mail service with the best privacy.
> 
> Every user has different requirements, but a notification system is a core feature for 95% of the users, thus cannot be sacrificed.

___

The point being made is not that ProtonMail is a bad service or that Google Notifications should be dropped overall. It is that it would be cool to also have a version for those 5% of customers, who don't use Google Notification Services. This might sound absourd, to have a different version for such a small group of your customers, I agree, but is important to recognise who those people are: They are the people who use your service primarily for privacy reasons and I think those are the people most willing to advocate for your company.
So maybe consider this :)

**Comment by ksdrummer** [comment]
I must add that these people are the target audience of protonmail since PM is advertising so much on privacy.
People who are cool with using google services are using gmail

**Comment by bartbutler** [comment]
Just to be clear, we're going to do this and have put a bunch of work into it already, but we don't have a public timeline as of yet.

**Comment by Ammako** [comment]
> @4face-studi0 since you ask, I use lineageOS with microG and so I don't have push notifications.

@ksdrummer but microG doesn't prevent you from having push notifications, in fact one of its main features is to enable Google FCM without using proprietary play services.

**Comment by ksdrummer** [comment]
@Ammako 
Yes I know but you don't have to use them if you don't want to. It is my choice to not use google push notification and microG made possible by microG.
Anyways, a google account is still needed to use GPN so why using microG if you sign in with a google account ? I mean if you're ok with that just use google like everyone else, but please let people who don't want to use any google service do it.

**Comment by Ammako** [comment]
> Anyways, a google account is still needed to use GPN

That is not the case!

**Comment by ksdrummer** [comment]
@Ammako Really ? that's a great news ! How do you do that ?

**Comment by Ammako** [comment]
You just have to enable Google device registration, and enable Cloud Messaging. Recent versions of microG GmsCore are a bit broken though, you might need to downgrade to 0.2.15 for everything to properly register. You can update back to the newest version afterwards, it'll continue working.

Supposedly, all identifying bits are removed from registration data, aside from Google account name, but if you aren't even using a Google account then you don't even have a Google account name to begin with. And the way Protonmail handles notifications, I'm pretty sure the google cloud notification ping is just a placeholder notification that tells the app to fetch new messages, and then the app fetches new messages and displays a notification with the email title and sender. So Google can know when you receive emails, but they can't know what they are.

It'll be great once Protonmail can add their own push notification system, then I can turn off device registration and cloud messaging and never look back. But for the time being, I think this is a good enough compromise.
